Mathlib/Analysis/Calculus/Taylor.lean

```diff
@@ -209,6 +212,82 @@ theorem hasDerivWithinAt_taylorWithinEval_at_Icc {f : ℝ → E} {a b t : ℝ} (
   hasDerivWithinAt_taylorWithinEval (uniqueDiffOn_Icc hx t ht) (uniqueDiffOn_Icc hx)
     self_mem_nhdsWithin ht rfl.subset hf (hf' t ht)
 
+/-- Calculate the derivative of the Taylor polynomial with respect to `x`. --/
+theorem hasDerivAt_taylorWithinEval_succ {x₀ x : ℝ} {s : Set ℝ}
+    (hs : UniqueDiffOn ℝ s) (hx₀ : x₀ ∈ s) (f : ℝ → E) (n : ℕ) :
+    HasDerivAt (taylorWithinEval f (n + 1) s x₀)
+      (taylorWithinEval (derivWithin f s) n s x₀ x) x := by
+  change HasDerivAt (fun x ↦ taylorWithinEval f _ s x₀ x) _ _
+  simp_rw [taylor_within_apply]
+  have : ∀ (i : ℕ) {c : ℝ} {c' : E},
+      HasDerivAt (fun x ↦ (c * (x - x₀) ^ i) • c') ((c * (i * (x - x₀) ^ (i - 1) * 1)) • c') x :=
+    fun _ _ ↦ hasDerivAt_id _ |>.sub_const _ |>.pow _ |>.const_mul _ |>.smul_const _
+  apply HasDerivAt.sum (fun i _ => this i) |>.congr_deriv
+  rw [Finset.sum_range_succ', Nat.cast_zero, zero_mul, zero_mul, mul_zero, zero_smul, add_zero]
+  apply Finset.sum_congr rfl
+  intro i _
+  rw [← iteratedDerivWithin_succ' hs hx₀]
+  congr 1
+  field_simp [Nat.factorial_succ]
+  ring
+
+/-- **Taylor's theorem** as a limit. -/
+theorem taylor_tendsto {f : ℝ → ℝ} {x₀ : ℝ} {n : ℕ} {s : Set ℝ}
+    (hs : UniqueDiffOn ℝ s) (hsx₀ : s ∈ 𝓝 x₀) (hf : ContDiffOn ℝ n f s) :
```

Comment:
You shouldn't need `hs` here: since `s` in a neighborhood of `x_0`, it contains a small interval around `x_0`, so there are is no issue of lack of differentiability there.

On the other hand, your version of the theorem does not apply to `[x_0, x_0+1]` for instance (i.e., it does not allow a one-sided version), which is a little bit sad. Does your proof work if you assume that `s` is convex and contains `x_0`, and that the limit is taken along ` 𝓝[s] x₀`? This would cover both the two-sided case that you are giving, but also the left and right one-sided versions.

Reply:
I've got a proof for convex `s` now. Thanks for the suggestion!